Adds interactive and output path options to model register

rikkarth opened the pull request:
### Description
Enables interactive mode and custom output path selection when registering a model via the CLI. Improves user experience by allowing prompts for missing parameters and adds validation for required arguments in non-interactive mode. Updates tests to cover new options.

Interactive mode, will prompt you for missing options
```bash
opensearch-ml model register -i
```

If no `-i` or `--interactive` flag is provided, by default non-interactive mode will be run
```bash
opensearch-ml model register \
--connectorId "$CONNECTOR_ID" \
--name "Example embedding" \
--description "Example embedding model connector" \
--outputPath "/path/to/file" # new option for improved automation
```

In non-interactive mode, each specific missing parameter will be listed and the application will be stopped until all requirements are met.
```
opensearch-ml model register

Starting model registration...

Setup configuration loaded successfully from /home/rikkarth/repoland/opensearch-py-ml/test_config.yml
Missing required parameters: 
        --name
        --description
        --connectorId
```

### Issues Resolved
- https://github.com/opensearch-project/opensearch-py-ml/issues/590
 
### Follow ups
Both predict and connector registration still do not support non-interactive modes. I will address this in two seperate PRs so it can be more digestable. In the meantime both predict and connector logic have been made backwards compatible with their existing implementation.

### Check List
- [x] New functionality includes testing.
  - [x] Existing tests have been reviewed and aligned with the new functionality.
  - [ ] All tests pass (could not run pipeline on github, locally all cli package tests pass)
- [x] New functionality has been documented.
  - [x] New functionality has docstrings added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

rikkarth committed:
Adds interactive and output path options to model register

Enables interactive mode and custom output path selection when registering a model via the CLI. Improves user experience by allowing prompts for missing parameters and adds validation for required arguments in non-interactive mode. Updates tests to cover new options.

Signed-off-by: Ricardo Mendes <rikkarth@pm.me>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth committed:
Add backwards compatibility with connector logic.

This will ensure the connector logic which still does not support direct connector registration still works as before.

Signed-off-by: Ricardo Mendes <rikkarth@pm.me>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth committed:
Add backwards compatibility with predict logic

This will ensure the predict logic which still does not support direct prediction still works as before.

Signed-off-by: Ricardo Mendes <rikkarth@pm.me>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth committed:
Add docstrings to output path and interactive mode

Signed-off-by: Ricardo Mendes <rikkarth@pm.me>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth committed:
nox format alignment

Signed-off-by: Ricardo Mendes <rikkarth@pm.me>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth committed:
Updates CLI docs with interactive and output options

Clarifies model registration syntax by adding interactive mode (-i) and output file path (-o) options.
Improves usage examples and option descriptions for better user guidance.

Signed-off-by: Ricardo Mendes <rikkarth@pm.me>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth committed:
nox lint and format alignment

Signed-off-by: Ricardo Mendes <rikkarth@pm.me>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth committed:
update CHANGELOG.md

Signed-off-by: Ricardo Mendes <rikkarth@pm.me>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth commented:
@nathaliellenaa Thank you so much for taking the time to review my PR. I have applied the changes you mentioned.

> lint checker is failing, you can run nox -s format to fix the formatting and run nox -s lint to verify it

Done

> Add your changes to the [CHANGELOG](https://github.com/opensearch-project/opensearch-py-ml/blob/main/CHANGELOG.md)

Done

> Can you also update our documentation to include these new changes https://github.com/opensearch-project/opensearch-py-ml/blob/main/docs/source/cli/register_model.rst

Done

nathaliellenaa reviewed (commented):
Overall changes looks good! Added one small comment. Can you also rebase with main and resolve the conflict?

rikkarth committed:
Clarifies output option name in CLI usage docs

Signed-off-by: Ricardo Mendes <dev@rikkarth.com>
Signed-off-by: rikkarth <dev@rikkarth.com>

rikkarth commented:
> Overall changes looks good! Added one small comment. Can you also rebase with main and resolve the conflict?

Hi @nathaliellenaa I performed the change as you suggested and rebased the branch.

Thank you so much for the review.

nathaliellenaa reviewed (approved):
Thanks @rikkarth for the PR, LGTM

rikkarth commented:
Hey, let me know if there's anything preventing this from being merged. Thank you.
